sigstore/_internal/trust.py

```diff
@@ -217,6 +220,102 @@ def __str__(self) -> str:
         return self.value
 
 
+class CertificateAuthority:
+    """
+    Certificate Authority used in a Trusted Root configuration.
+    """
+
+    def __init__(self, inner: _CertificateAuthority):
+        """
+        Construct a new `CertificateAuthority`.
+
+        @api private
+        """
+        self._inner = inner
+        self._leaf: Certificate | None = None
+        self._intermediates: list[Certificate] = []
+        self._root: Certificate | None = None
+        self._verify()
+
+    @classmethod
+    def from_json(cls, path: str) -> CertificateAuthority:
+        """
+        Create a CertificateAuthority directly from JSON.
+        """
+        inner = _CertificateAuthority().from_json(Path(path).read_bytes())
+        return cls(inner)
+
+    def _verify(self) -> None:
+        """
+        Verify and load the certificate authority.
+        """
+        chain_length = len(self._inner.cert_chain.certificates)
+        self._intermediates = []
+
+        for idx, cert in enumerate(self._inner.cert_chain.certificates):
+            certificate = load_der_x509_certificate(cert.raw_bytes)
+            if idx == 0 and not cert_is_root_ca(certificate):
+                self._leaf = certificate
+            elif idx < chain_length - 1:
+                self._intermediates.append(certificate)
+            elif idx == chain_length - 1 and cert_is_root_ca(certificate):
+                self._root = certificate
+
+        if not self._root:
+            raise Error("missing root certificate in CA")
+
+    @property
+    def validity_period_start(self) -> datetime | None:
+        """
+        Validity period start.
+        """
+        return self._inner.valid_for.start
+
+    @property
+    def validity_period_end(self) -> datetime | None:
+        """
+        Validity period end.
+        """
+        return self._inner.valid_for.end
+
+    @property
+    def root(self) -> Certificate | None:
+        """
+        Root certificate of the authority.
+        """
+        return self._root
+
+    @property
+    def intermediates(self) -> list[Certificate]:
+        """
+        List of intermediates certificates.
+        """
+        return self._intermediates
```

Comment:
Related: I think we can remove both of these, and allow access only through the `certificates(...)` method. FWICT they're only used in tests anyways.